### PR TITLE
Closed loop 2067079 2063691: Test s3 routes update/reconcile

### PR DESCRIPTION
**Pull request by parikshithb**
Validates  2067079 2063691

Added bz marker for 2029298 in object expiration test.
Signed-off-by: Parikshith Byregowda <pbyregow@redhat.com>

### CODE REVIEW

File: tests/manage/mcg/test_s3_routes.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                        sleep(RECONCILE_WAIT)
+                except KeyError:
+                    logger.info(
+                        "disableRoute does not exist in storage cluster, no need to revert"
+                    )
+                rgw_route_obj = ocp.OCP(
+                    kind=constants.ROUTE,
+                    namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+                )
+                assert rgw_route_obj.is_exist(
```
Comment (mashetty330) [inline tests/manage/mcg/test_s3_routes.py:67]: assert with the a failure message?

File: tests/manage/mcg/test_s3_routes.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                namespace=defaults.ROOK_CLUSTER_NAMESPACE,
+                resource_name="s3",
+            )
+            if (
+                nb_s3_route_obj.data["spec"]["tls"]["insecureEdgeTerminationPolicy"]
+                == "Redirect"
+            ):
+                s3_route_param = '{"spec":{"tls":{"insecureEdgeTerminationPolicy":"Allow","termination":"reencrypt"}}}'
+                nb_s3_route_obj.patch(params=s3_route_param, format_type="merge")
+
```
Comment (mashetty330) [inline tests/manage/mcg/test_s3_routes.py:45]: you can verify revert patch has applied to s3 route

**Review by mashetty330 — COMMENTED** [review]
Few minor comments

File: tests/manage/mcg/test_s3_routes.py (hunk 2 above)
Comment (parikshithb) [reply tests/manage/mcg/test_s3_routes.py:46]: ack

File: tests/manage/mcg/test_s3_routes.py (hunk 1 above)
Comment (parikshithb) [reply tests/manage/mcg/test_s3_routes.py:77]: added

**Review by mashetty330 — APPROVED** [review]

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**